int32 id
string type
---
int32[] ids
